packs/actions/solicitation.ts

```diff
@@ -0,0 +1,139 @@
+import type { AppContext } from "$store/apps/site.ts";
+import { createClient } from "https://esm.sh/v135/@supabase/supabase-js@2.7.0";
+import {
+  RESPONSE_RISK3_APPROVED,
+  SOLICITATION_ENTITY_NAME,
+  STATUS_ENUM_ABLE,
+  STATUS_ENUM_ACCOUNT_OPENING,
+  STATUS_ENUM_DISAPPROVED,
+  STATUS_ENUM_IN_OPERATION,
+  STATUS_ENUM_SIGNATURE,
+  STATUS_ENUM_SUSPENDED,
+} from "deco-sites/niivu-bank/packs/utils/constants.ts";
+
+interface Fields {
+  phone: string;
+  zip_code: string;
+  street: string;
+  number: string;
+  complement?: string;
+  city: string;
+  state: string;
+  email: string;
+  rg?: string;
+  full_name?: string;
+  cpf?: string;
+  business_name?: string;
+  cnpj?: string;
+}
+
+interface Entity extends Fields {
+  credit_status: boolean;
+  status: StatusEnum;
+}
+
+export type StatusEnum =
+  | typeof STATUS_ENUM_ACCOUNT_OPENING
+  | typeof STATUS_ENUM_SIGNATURE
+  | typeof STATUS_ENUM_ABLE
+  | typeof STATUS_ENUM_IN_OPERATION
+  | typeof STATUS_ENUM_SUSPENDED
+  | typeof STATUS_ENUM_DISAPPROVED;
+
+interface Props extends Fields {
+  type: "CPF" | "CNPJ";
+}
+
+export default async function loader(
+  props: Props,
+  _req: Request,
+  ctx: AppContext,
+) {
+  const { supaBase, risk3 } = ctx;
+  const { clientRisk3, password, username, product } = risk3;
+  if (!supaBase) {
+    return "You must provide supaBase fields in site.ts";
+  }
+
+  const client = createClient(supaBase.url!, supaBase.token!);
+
+  const { type, ...rest } = props;
+
+  if (type !== "CPF" && type !== "CNPJ") {
+    return "error, wrong type.";
+  }
+
+  const usernameStr = typeof username === "string" ? username : username?.get();
+  const passwordSrt = typeof password === "string" ? password : password?.get();
+
+  if (!usernameStr || !passwordSrt) {
+    return "error, credenciais de Usuário e senha não foram definidas.";
+  }
+
+  const res = await clientRisk3["POST /api/v0/login"]({
+    username: usernameStr,
+    password: passwordSrt,
+  }, {});
+  const response = await res.json();
+
+  const { data, status, message } = response;
+
+  if (status === "error" && !data) {
+    return message;
+  }
+
+  if (type === "CPF") {
+    const res = await clientRisk3["POST /api/v0/analises/cpf"]({
+      "Venidera-AuthToken": `Bearer ${data.token}`,
```

Comment:
Reutilzia o que for possivel
```suggestion
   const AUTH_TOKEN = `Bearer ${data.token}`
      ...
      "Venidera-AuthToken": authToken,
```